packages/core/src/deployment.ts

```diff
@@ -17,7 +17,7 @@ export async function resumeOrDeploy<T extends Deployment>(
 ): Promise<T> {
   // If there is a deployment stored, we look its transaction up. If the
   // transaction is found, the deployment is reused.
-  if (cached !== undefined) {
+  if (cached?.txHash !== undefined) {
```

Comment:
The result of this is that if `txHash` is undefined the contract is redeployed. But if `txHash` is undefined it's because the contract was migrated from a CLI project, and we can probably reuse it.

The body of this `if` will check that the transaction has been mined. We can't do that without `txHash`. We can probably assume that the tx has been mined, or we could try to check that there is code at the address.